modules/openxr/openxr_api.cpp

```diff
@@ -2376,6 +2382,26 @@ void OpenXRAPI::end_frame() {
 		}
 	}
 
+	Rect2i new_render_region = Rect2i(Point2i(0, 0), render_state.main_swapchain_size);
+	if (render_region != Rect2i()) {
+		new_render_region = render_region;
+	}
```

Comment:
Can be replaced by the following:
```
Rect2i new_render_region = (render_region != Rect2i()) ? render_region : Rect2i(Point2i(0, 0), render_state.main_swapchain_size);
```

In addition:
- The [`Rect2i` documentation](https://docs.godotengine.org/en/stable/classes/class_rect2i.html) states that `Rect2i() == false`, so we can replace `render_region != Rect2i()` with `render_region != false`
- `Point2i(0, 0)` can be replaced by `Vector2i.ZERO`
- Do we expect `Rect2i(Point2i(0, 0), render_state.main_swapchain_size)` to change every frame?

Reply:
I've made the suggested replacement, thanks! As to your other points, the `Rect2i() == false` and `Vector2i.ZERO` suggestions I believe would only work in editor?

`Rect2i(Point2i(0, 0), render_state.main_swapchain_size)` would typically remain the same, it would only be resized if a new swapchain was created (with a different size). Maybe on swapchain creation I could just store the size in a variable rather than creating this every frame.